### ISSUE TITLE
Reduce References to cli.Context of the Sharding Node in its Services

### ISSUE THREAD

**Issue by rauljordan** [prysmaticlabs/geth-sharding#132]
Hi all,

As pointed out by @prestonvanloon in the comments in #127, we have a bunch of places where we reference the sharding node's .Context(). It would be best if we can simplify this by instead just passing in the values the services will need. Instead of calling .Context() to fetch a cli flag, we could just pass in that cli flag instead. This makes it clear to the reader what cli args each service will need.

If anyone wants to learn more about this or understand the context of this discussion, reference this issue over out [Gitter](https://gitter.im/prysmaticlabs/geth-sharding) channel.

**Comment by prestonvanloon** [comment]
To shine more light on this: my specific suggestion was to add abstraction around flag access for the users of the sharding `node`.

I.e. change this
```
actor := n.Context().GlobalString(utils.ActorFlag.Name)
```
to this
```
actor := n.Actor()
```

Why? It's easier to use, consume, and understand for the bit of code that needs to know what type of actor to operate in this specific example. It probably makes it easier to test as well. 

There may be only one instance of this specific example, but it would be worth looking at other ways to improve our interfaces as part of this issue.

**Comment by rauljordan** [comment]
Great point, this will reduce the need for certain imports and makes the code a lot more clear. I already started fixing this in #137 where I use `n.DataDir()` as an abstraction over the cli context. This, along with the DepositFlag are the only two scenarios here but we will keep this in mind moving forward. I'll fix the DepositFlag situation in a different PR.

**Comment by rauljordan** [comment]
Closed by #149
